LogiusWebApp/src/main/java/org/verapdf/crawler/logius/resources/UserResource.java

```diff
@@ -60,8 +68,37 @@ public ResponseEntity getAllUsers(
     }
 
     @PostMapping("/signup")
-    public ResponseEntity registerUser(@Valid @RequestBody UserDto signUpRequest) {
-        userService.save(signUpRequest);
+    public ResponseEntity registerUser(@RequestBody @Valid UserDto signUpRequest) {
+        User user = userService.save(signUpRequest);
+        sendEmail.sendEmailConfirm(tokenService.encode(user), user.getEmail());
         return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
     }
+
+    @PostMapping("/email-confirm")
+    public String emailConfirm(@RequestParam(value = "token") String token) {
```

Comment:
move token to general security layer validation with additional scope (ACTIVATION)